strawberryfields/device.py

```diff
@@ -45,9 +46,28 @@ def __init__(self, spec: Mapping[str, Any], cert: Optional[Mapping[str, Any]] =
                 f"Device specification is missing the following keys: {sorted(missing_keys)}"
             )
 
-        self._spec = spec
+        self._spec = self.assert_spec(spec)
         self._certificate = cert
 
+    @staticmethod
+    def assert_spec(spec: Mapping[str, Any]):
```

Comment:
I wonder if it is clearer to call this `validate_target`?

Reply:
Yes, that's probably better. I was thinking this as a more general spec validation, but you're correct. `validate_target` makes more sense the way it's written now.